src/PKSim.CLI/Program.cs

```diff
@@ -5,74 +5,82 @@
-            .WithParsed<JsonRunCommand>(startCommand)
-            .WithParsed<SnapshotRunCommand>(startCommand)
-            .WithParsed<ExportRunCommand>(startCommand)
-            .WithParsed<QualificationRunCommand>(startCommand)
-            .WithNotParsed(err => _valid = false);
+      Parser.Default.ParseArguments<JsonRunCommand, SnapshotRunCommand, ExportRunCommand, QualificationRunCommand>(args)
+         .WithParsed<JsonRunCommand>(startCommand)
+         .WithParsed<SnapshotRunCommand>(startCommand)
+         .WithParsed<ExportRunCommand>(startCommand)
+         .WithParsed<QualificationRunCommand>(startCommand)
+         .WithNotParsed(err => _valid = false);
 
-         if (!_valid)
-            return (int) ExitCodes.Error;
+      if (!_valid)
+        return (int)ExitCodes.Error;
 
-         return (int) ExitCodes.Success;
-      }
+      return (int)ExitCodes.Success;
+    }
 
-      private static void startCommand<TRunOptions>(CLICommand<TRunOptions> command)
-      {
-         var logger = initializeLogger(command);
-         if(command.LogCommandName)
-            logger.AddInfo($"Starting {command.Name.ToLower()} run");
+    private static void startCommand<TRunOptions>(CLICommand<TRunOptions> command)
+    {
+      var logger = initializeLogger(command);
+      if (command.LogCommandName)
+        logger.AddInfo($"Starting {command.Name.ToLower()} run");
 
-         logger.AddDebug($"Arguments:\n{command}");
-         ApplicationStartup.Start();
-         var runner = IoC.Resolve<IBatchRunner<TRunOptions>>();
-         try
-         {
-            runner.RunBatchAsync(command.ToRunOptions()).Wait();
-         }
-         catch (Exception e)
-         {
-            logger.AddException(e);
-            _valid = false;
-         }
-
-         if (command.LogCommandName)
-            logger.AddInfo($"{command.Name} run finished");
+      logger.AddDebug($"Arguments:\n{command}");
+      ApplicationStartup.Start();
+      var runner = IoC.Resolve<IBatchRunner<TRunOptions>>();
+      try
+      {
+        runner.RunBatchAsync(command.ToRunOptions()).Wait();
       }
-
-      private static ILogger initializeLogger(CLICommand runCommand)
+      catch (Exception e)
       {
-         var loggerFactory = IoC.Resolve<ILoggerFactory>();
+        logger.AddException(e);
+        _valid = false;
+      }
 
-         loggerFactory
-            .AddConsole(runCommand.LogLevel);
+      if (command.LogCommandName)
+        logger.AddInfo($"{command.Name} run finished");
+    }
 
-         if (!string.IsNullOrEmpty(runCommand.LogFileFullPath))
-            loggerFactory.AddFile(runCommand.LogFileFullPath, runCommand.LogLevel, runCommand.AppendToLog);
+    private static IOSPLogger initializeLogger(CLICommand runCommand)
+    {
 
-         return IoC.Resolve<ILogger>();
-      }
-   }
+      var loggerCreator = IoC.Resolve<ILoggerCreator>();
```

Comment:
This is werid. LoggerCreator is just created locally and then not used.  
Should we rename it to something like LoggerConfigurator or sthg like this?

Reply:
@msevestre It is used. It will be injected to the PKSimLogger and it should be a singleton. We might have forgotten to register it as a singleton though for the contract used in this case. Maybe @georgeDaskalakis could take a look into this? Otherwise I'll do it. About naming, well we could rename it but it is not just configuring loggers, it is actually creating the loggers and keeping them in a dictionary based on the category. We are not keeping the LoggerFactory, nor the LoggingBuilder alive anymore since this is not the way to go in the new version. Instead we keep the loggers alive while the app is running. This loggerCreator has then a ConcurrentDictionary from category names to loggers and also knows how to create new loggers in case of first use.

Reply:
@msevestre @georgeDaskalakis About _"Maybe @georgeDaskalakis could take a look into this? Otherwise I'll do it"_ it is already registered as a singleton in PKSim.Infrastructure.InfrastructureRegister